backend/src/service/scan-mode.service.ts

```diff
@@ -0,0 +1,73 @@
+import JoiValidator from '../web-server/controllers/validators/joi.validator';
+import ScanModeRepository from '../repository/scan-mode.repository';
+import OIAnalyticsMessageService from './oia/oianalytics-message.service';
+import { ScanModeCommandDTO, ValidatedCronExpression } from '../../../shared/model/scan-mode.model';
+import { scanModeSchema } from '../web-server/controllers/validators/oibus-validation-schema';
+import SouthCacheRepository from '../repository/south-cache.repository';
+import { ScanMode, ScanModeCommand } from '../model/scan-mode.model';
+import { validateCronExpression } from './utils';
+import OIBusEngine from '../engine/oibus-engine';
+
+export default class ScanModeService {
+  constructor(
+    protected readonly validator: JoiValidator,
+    private scanModeRepository: ScanModeRepository,
+    private southCacheRepository: SouthCacheRepository,
+    private oIAnalyticsMessageService: OIAnalyticsMessageService,
+    private oibusEngine: OIBusEngine
+  ) {}
+
+  findAll(): Array<ScanMode> {
+    return this.scanModeRepository.findAll();
+  }
+
+  findById(id: string): ScanMode | null {
+    const scanMode = this.scanModeRepository.findById(id);
+    if (scanMode) {
+      return scanMode;
+    } else {
+      return null;
+    }
+  }
+
+  async create(command: ScanModeCommandDTO): Promise<ScanMode> {
+    await this.validator.validate(scanModeSchema, command);
+    const scanMode = this.scanModeRepository.create(command as ScanModeCommand);
+    this.oIAnalyticsMessageService.createFullConfigMessage();
+    return scanMode;
+  }
+
+  async update(scanModeId: string, command: ScanModeCommandDTO): Promise<void> {
+    await this.validator.validate(scanModeSchema, command);
+    const oldScanMode = this.scanModeRepository.findById(scanModeId);
+    if (!oldScanMode) {
+      throw new Error(`Scan mode ${scanModeId} not found`);
+    }
+    this.scanModeRepository.update(scanModeId, command as ScanModeCommand);
+    const newScanMode = this.scanModeRepository.findById(scanModeId)!;
+    if (oldScanMode.cron !== newScanMode.cron) {
+      await this.oibusEngine.updateScanMode(newScanMode);
+    }
+    this.oIAnalyticsMessageService.createFullConfigMessage();
+  }
+
+  async delete(scanModeId: string): Promise<void> {
+    const scanMode = this.scanModeRepository.findById(scanModeId);
+    if (!scanMode) {
+      throw new Error(`Scan mode ${scanModeId} not found`);
+    }
+
+    this.scanModeRepository.delete(scanModeId);
+    this.southCacheRepository.deleteAllByScanMode(scanModeId);
```

Comment:
do you have a foreign key? if yes then maybe the order matters not sure but this require a closer look